socket/index.js

```diff
@@ -1,185 +1,152 @@
-const PrivateRoomManager = require("./private_room")
-class IDManager {
-  constructor() {
-    this.ids = {};
-  }
-  add(userId, id) {
-    this.ids[userId] = id;
-  };
-
-  remove(id) {
-    for (let user in this.ids)
-      if (this.ids[user] == id) {
-        this.ids[user] = null;
-        delete this.ids[user]
-      }
-  };
-
-  get(userId) {
-    return this.ids[userId];
-  };
-  get_reversed(id) {
-    for (let user in this.ids)
-      if (this.ids[user] == id) {
-        return user;
-      }
-    return null;
-  }
-}
 
 class Socket {
   constructor() {
     console.log("Initializing socket")
     this.io = null;
-    this.UserSocket = new IDManager(); // userId - socketId
-    this.users = {};
-    this.socketRoomMap = {};
-    this.privateRoomManager = new PrivateRoomManager();
+    this.userSocketIDMapper = {}; // userID - socketID
   }
 
-  getPartnerPeerJS(peerid) {
-    for (let peer of this.peers) {
-      if (peerId == peer[0]) {
-        return peer[1]
-      }
-      if (peerId == peer[1]) {
-        return peer[0]
-      }
-    }
-    return null;
-  }
   init(httpServer) {
 
     this.io = socket(httpServer, {
       cors: {
-        origins: ['http://localhost:3000']
+        origins: JSON.parse(process.env.SOCKET_ORIGINS)
       }
     });
     console.log('Socket is ready!')
 
     this.io.on("connection", (socket) => {
-      //when connect
       console.log("A user connected.");
 
       //take userId and socketId from user
-      socket.on("addUser", (payload) => {
-        console.log("add user: ", this.UserSocket.ids)
-        this.UserSocket.add(payload.userId, socket.id);
+      socket.on("add-user", (payload) => {
+        console.log("add user: ", this.userSocketIDMapper.ids)
+        this.userSocketIDMapper[payload.userId] = socket.id;
+      });
+
+      socket.on("send-message", (message) => {
+        console.log("new message: ", message)
+        const recipientSocket = this.userSocketIDMapper[message.recipientId];
+        const senderSocket = this.userSocketIDMapper[message.senderId];
+        try {
+          if (recipientSocket)
+            this.io.to(recipientSocket).emit("new-message", message);
+          if (senderSocket)
+            this.io.to(senderSocket).emit("new-message", message);
+        } catch (error) {
+          console.log(error)
+        }
       });
-      const events = ['call-request', 'accepted', 'rejected', 'cancel-call']
+
+      socket.on("disconnect", () => {
+        console.log("a user disconnected!");
+        this.userSocketIDMapper.remove(socket.id);
+      });
+    });
+    return this.io;
+  }
+
+  sendToUser(userId, event, payload) {
+    try {
+      const userSocket = this.userSocketIDMapper[userId];
+      if (userSocket) {
+        console.log("Sending event ", event, "payload: ", payload)
+        this.io.to(userSocket).emit(event, payload)
+      }
+    } catch (error) {
+      console.log(error)
+    }
+  }
+}
+
+class CallableSocket extends Socket {
```

Comment:
Creating a subclass that helps the original Socket obey the single responsibility principle